apps/web/src/app/[locale]/challenge/_components/comments/comment-input.tsx

```diff
@@ -14,65 +12,38 @@ import { singleFieldSchema, type SingleFieldSchema } from '~/utils/zodSingleStri
 interface Props {
   mode: 'create' | 'edit' | 'reply';
   onCancel?: () => void;
-  onChange: (text: string) => void;
-  value: string;
   placeholder?: string;
-  onSubmit: () => Promise<void>;
+  onSubmit: (text: string) => Promise<void>;
+  defaultValue?: string;
 }
 
-export function CommentInput({ mode, onCancel, onChange, value, placeholder, onSubmit }: Props) {
+export function CommentInput({ mode, onCancel, placeholder, onSubmit, defaultValue }: Props) {
   const { data: session } = useSession();
-  const { toast } = useToast();
   const [commentMode, setCommentMode] = useState<'editor' | 'preview'>('editor');
-  const [isSubmitting, setIsSubmitting] = useState(false);
+
   const textAreaRef = useRef<HTMLTextAreaElement>(null);
 
   const form = useForm<SingleFieldSchema>({
     resolver: zodResolver(singleFieldSchema),
-    defaultValues: { text: '' },
+    defaultValues: { text: defaultValue ?? '' },
   });
   const formValid = form.formState.isValid;
-
+  const value = form.watch('text');
   useAutosizeTextArea(textAreaRef, value, commentMode);
 
-  const handleEnterKey = async () => {
-    if (!session?.user) {
-      toast({
-        variant: 'destructive',
-        title: 'You need to be logged in to comment.',
-        action: <ToastAction altText="Dismiss">Dismiss</ToastAction>,
-      });
-
-      return;
-    }
-    setIsSubmitting(true);
-    await onSubmit();
-    setIsSubmitting(false);
-  };
-
-  const submitComment = async () => {
+  const submitComment = async ({ text }: { text: string }) => {
     try {
-      if (!session?.user) {
```

Comment:
(non-blocking) question: do we still capture the logged-in requirement somewhere else?

Reply:
Yes, it was duplicated logic, that's why I removed it